functests/gatekeeper/gatekeeper.go

```diff
@@ -0,0 +1,151 @@
+package gatekeeper
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	dynamic "github.com/openshift-kni/cnf-features-deploy/functests/utils/dynamic"
+)
+
+type metadataInfo struct {
+	kind  string
+	key   string
+	value string
+}
+
+type assignMetadataInfo struct {
+	name     string
+	kinds    []string
+	scope    string
+	metadata metadataInfo
+}
+
+var _ = Describe("gatekeeper", func() {
+	// we make the interface key to a var for runtime key check
+	// if we used the string directly a typo will lead to a runtime error
+	// typos in var names will make a compile error
+	assignMetadata := "assignmetadata"
+	pods := "pods"
+	// create the dynamic client
+	client := dynamic.New("")
```

Comment:
On first look I assumed this is the standard dynamic client (of the client-go libs). Maybe we could go with something like testClient to avoid confustion (and packge naming colision)?